Meet error about `...` operator

mini example: 
```ts
class Foo {
    data(): Array<u8> {
        return [];
    }
}
export class Bar extends Foo {
    data(): Array<u8> {
        return [...super.data()];
    }
}
```

The error:
```
▌ Whoops, the AssemblyScript compiler has crashed during compile :-(
▌ 
▌ Here is a stack trace that may or may not be useful:
▌ 
▌ AssertionError: assertion failed
▌     at i.assert (C:\Users\lenovo\AppData\Roaming\npm\node_modules\assemblyscript\dist\assemblyscript.js:7:737243)
▌     at f.compileUnaryPrefixExpression (C:\Users\lenovo\AppData\Roaming\npm\node_modules\assemblyscript\dist\assemblyscript.js:7:331940)
▌     at f.compileExpression (C:\Users\lenovo\AppData\Roaming\npm\node_modules\assemblyscript\dist\assemblyscript.js:7:238684)
▌     at f.compileArrayLiteral (C:\Users\lenovo\AppData\Roaming\npm\node_modules\assemblyscript\dist\assemblyscript.js:7:310195)
▌     at f.compileLiteralExpression (C:\Users\lenovo\AppData\Roaming\npm\node_modules\assemblyscript\dist\assemblyscript.js:7:306265)
▌     at f.compileExpression (C:\Users\lenovo\AppData\Roaming\npm\node_modules\assemblyscript\dist\assemblyscript.js:7:238319)
▌     at f.compileReturnStatement (C:\Users\lenovo\AppData\Roaming\npm\node_modules\assemblyscript\dist\assemblyscript.js:7:230012)
▌     at f.compileStatement (C:\Users\lenovo\AppData\Roaming\npm\node_modules\assemblyscript\dist\assemblyscript.js:7:223143)
▌     at f.compileStatements (C:\Users\lenovo\AppData\Roaming\npm\node_modules\assemblyscript\dist\assemblyscript.js:7:223935)
▌     at f.compileFunctionBody (C:\Users\lenovo\AppData\Roaming\npm\node_modules\assemblyscript\dist\assemblyscript.js:7:213143)
▌ 
▌ If it refers to the dist files, try to 'npm install source-map-support' and
▌ run again, which should then show the actual code location in the sources.
▌ 
▌ If you see where the error is, feel free to send us a pull request. If not,
▌ please let us know: https://github.com/AssemblyScript/assemblyscript/issues
▌ 
▌ Thank you!
```

--- a/NOTICE
+++ b/NOTICE
@@ -39,7 +39,7 @@ under the licensing terms detailed in LICENSE:
 * bnbarak <bn.barak@gmail.com>
 * Colin Eberhardt <colin.eberhardt@gmail.com>
 * Ryan Pivovar <ryanpivovar@gmail.com>
-* Roman F. <romdotdog@users.noreply.github.com>
+* Roman F. <70765447+romdotdog@users.noreply.github.com>
 * Joe Pea <trusktr@gmail.com>
 
 Portions of this software are derived from third-party works licensed under

--- a/NOTICE
+++ b/NOTICE
@@ -39,6 +39,7 @@ under the licensing terms detailed in LICENSE:
 * bnbarak <bn.barak@gmail.com>
 * Colin Eberhardt <colin.eberhardt@gmail.com>
 * Ryan Pivovar <ryanpivovar@gmail.com>
+* Roman F. <romdotdog@users.noreply.github.com>
 
 Portions of this software are derived from third-party works licensed under
 the following terms:

--- a/src/compiler.ts
+++ b/src/compiler.ts
@@ -9745,6 +9745,13 @@ export class Compiler extends DiagnosticEmitter {
       case Token.TYPEOF: {
         return this.compileTypeof(expression, contextualType, constraints);
       }
+      case Token.DOT_DOT_DOT: {
+        this.error(
+          DiagnosticCode.Not_implemented_0,
+          expression.range, "Spread operator"
+        );
+        return module.unreachable();
+      }
       default: {
         assert(false);
         return module.unreachable();

--- a/src/resolver.ts
+++ b/src/resolver.ts
@@ -1769,10 +1769,12 @@ export class Resolver extends DiagnosticEmitter {
         return type.intType;
       }
       case Token.DOT_DOT_DOT: {
-        this.error(
-          DiagnosticCode.Not_implemented_0,
-          node.range, "Spread operator"
-        );
+        if (reportMode == ReportMode.REPORT) {
+          this.error(
+            DiagnosticCode.Not_implemented_0,
+            node.range, "Spread operator"
+          );
+        }
         return null;
       }
       default: assert(false);

--- a/src/resolver.ts
+++ b/src/resolver.ts
@@ -1768,6 +1768,13 @@ export class Resolver extends DiagnosticEmitter {
         }
         return type.intType;
       }
+      case Token.DOT_DOT_DOT: {
+        this.error(
+          DiagnosticCode.Not_implemented_0,
+          node.range, "Spread operator"
+        );
+        return null;
+      }
       default: assert(false);
     }
     return null;

